src/oidc.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::Context;
+use tokio::task;
+
+use crate::db::oidc::GoogleProviderMetadata;
+
+use openidconnect::core::CoreClient;
+use openidconnect::reqwest::http_client;
+use openidconnect::{ClientId, ClientSecret, IssuerUrl, RedirectUrl, RevocationUrl};
+
+pub async fn oidc_client(
```

Comment:
This function and the rest of your changes still contain some unwraps. Please return `anyhow::Result` from your functions and use error handling with `?` [as described here](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html).